src/Graphics/Glyphs/TensorGlyphBuilder.h

```diff
@@ -40,37 +40,66 @@
 
 namespace SCIRun {
 namespace Graphics {
+struct EllipsoidPointParams
+{
+  double sinPhi;
+  double cosPhi;
+  double sinTheta;
+  double cosTheta;
+};
+
+struct SuperquadricPointParams
+{
+  double sinPhi;
+  double cosPhi;
+  double sinTheta;
+  double cosTheta;
+  double A;
+  double B;
+};
+
 class SCISHARE TensorGlyphBuilder
 {
 public:
   TensorGlyphBuilder(const Core::Geometry::Tensor& t, const Core::Geometry::Point& center);
   void scaleTensor(double scale);
   void reorderTensorValues(std::vector<Core::Geometry::Vector>& eigvecs, std::vector<double>& eigvals);
-  void makeTensorPositive();
+  void makeTensorPositive(bool reorder = false, bool makeGlyph = true);
   void normalizeTensor();
   void setColor(const Core::Datatypes::ColorRGB& color);
   void setResolution(double resolution);
   void generateSuperquadricTensor(GlyphConstructor& constructor, double emphasis);
   void generateEllipsoid(GlyphConstructor& constructor, bool half);
   void generateBox(GlyphConstructor& constructor);
 
+  Core::Geometry::Point evaluateSuperquadricPointLinear(const SuperquadricPointParams& params);
+  Core::Geometry::Point evaluateSuperquadricPointPlanar(const SuperquadricPointParams& params);
+  Core::Geometry::Point evaluateSuperquadricPoint(bool linear, const SuperquadricPointParams& params);
+  Core::Geometry::Point evaluateEllipsoidPoint(EllipsoidPointParams& params);
+
+  void computeTransforms();
+  void computeSinCosTable(bool half);
+  void postScaleTransorms();
+  Core::Geometry::Transform getTrans();
+  Core::Geometry::Transform getRotate();
+  Core::Geometry::Transform getScale();
+  bool isLinear();
+  void computeAAndB(double emphasis);
+  double getA();
+  double getB();
+  double computeSinPhi(int v);
+  double computeCosPhi(int v);
+  double computeSinTheta(int u);
+  double computeCosTheta(int u);
+  void setTensor(const Core::Geometry::Tensor &t);
+  Core::Geometry::Tensor getTensor() const;
+  std::vector<Core::Geometry::Vector> getEigenVectors();
+  std::vector<double> getEigenValues();
```

Comment:
Class is getting too big, but I'll let you split it up later.

Reply:
Okay, I'll address this with in the next PR with the new module.